lib/entities/provider_types.dart

```diff
@@ -48,61 +48,27 @@ extension ProviderTypeName on ProviderType {
 
 class ProvidersHelper {
   static List<ProviderType> getAvailableBuyProviderTypes(WalletType walletType) {
-    switch (walletType) {
-      case WalletType.nano:
-      case WalletType.banano:
-        return [ProviderType.askEachTime, ProviderType.onramper];
-      case WalletType.monero:
-        return [ProviderType.askEachTime, ProviderType.onramper, ProviderType.dfx];
-      case WalletType.bitcoin:
-      case WalletType.polygon:
-      case WalletType.ethereum:
-        return [
-          ProviderType.askEachTime,
-          ProviderType.onramper,
-          ProviderType.dfx,
-          ProviderType.robinhood,
-          ProviderType.moonpay,
-        ];
-      case WalletType.litecoin:
-      case WalletType.bitcoinCash:
-        return [ProviderType.askEachTime, ProviderType.onramper, ProviderType.robinhood, ProviderType.moonpay];
-      case WalletType.solana:
-        return [ProviderType.askEachTime, ProviderType.onramper, ProviderType.robinhood];
-      case WalletType.none:
-      case WalletType.haven:
-        return [];
+    final providers = <ProviderType>[];
+    for (final providerType in ProviderType.values) {
+      final dynamic p = getProviderTypeByType(providerType);
+      final supportedWalletTypes = p.getSupportedWalletTypes(true) as List<WalletType>;
```

Comment:
Why dynamic, make it `BuyProvider?` thus it will give you an error when directly calling getSupportedWalletTypes as p might be null
Modify it elsewhere as well

Reply:
Because it's not actually a BuyProvider instance that we're passing (because we wanted to avoid initializing the BuyProvider)

It's the type which inherits BuyProvider that we're passing

Afaik there's no way to do this properly without dynamic :/